iroh-util/src/lock.rs

```diff
@@ -1,79 +1,304 @@
+                .map_err(|e| LockError::Uncategorized { source: anyhow!(e) }),
+            Ok(true) => Err(LockError::Locked(self.path.clone())),
+            Err(e) => match e {
+                LockError::CorruptLock(_) => {
+                    // overwrite corrupt locks
+                    self.write().map_err(|e| LockError::Uncategorized {
+                        source: anyhow!("{}", e),
+                    })
+                }
+                e => Err(LockError::Uncategorized {
+                    source: anyhow!("{}", e),
+                }),
+            },
+        }
+    }
+
+    fn write(&mut self) -> AnyhowResult<()> {
+        // create lock. ensure path to lock exists
+        std::fs::create_dir_all(&crate::iroh_data_root()?)?;
+        let mut file = File::create(&self.path)?;
+        let pid = sysinfo::get_current_pid().unwrap();
+        file.write_all(pid.to_string().as_bytes())?;
+        self.lock = Some(pid);
+        Ok(())
+    }
 
-        file_guard::lock(file, Lock::Exclusive, 0, 1)
-            .map(|lock| self.lock = Some(lock))
-            .map_err(|err| err.into())
+    pub fn destroy_without_checking(&self) -> AnyhowResult<()> {
+        std::fs::remove_file(&self.path).map_err(|e| e.into())
     }
 }
 
+impl Drop for ProgramLock {
+    fn drop(&mut self) {
+        if self.lock.is_some() {
+            if let Err(err) = std::fs::remove_file(&self.path) {
+                warn!("removing lock: {}", err);
+            }
+        }
+    }
+}
+
+fn process_is_running(pid: Pid) -> AnyhowResult<bool> {
+    // existentialism is sometimes counterproductive
+    let this_pid = sysinfo::get_current_pid().unwrap();
+    if pid == this_pid {
+        return Ok(true);
+    }
+
+    // TODO(b5) - docs say we shouldn't be allocating on each call like this:
+    // https://docs.rs/sysinfo/0.26.5/sysinfo/index.html#usage
+    // I'm suspicious this pattern might be alright. Seems the underlying lib
+    // doesn't do much hydrating, but System may be a large memory allocation
+    let mut s = System::new();
+    if !s.refresh_process(pid) {
+        return Ok(false);
+    }
+
+    match s.process(pid) {
+        Some(process) => {
+            match process.status() {
+                // see https://docs.rs/sysinfo/0.26.5/sysinfo/enum.ProcessStatus.html for platform-specific details
+                ProcessStatus::Idle => Ok(true),
+                ProcessStatus::Run => Ok(true),
+                ProcessStatus::Sleep => Ok(true),
+                ProcessStatus::Waking => Ok(true),
+
+                ProcessStatus::Stop => Ok(false),
+                ProcessStatus::Zombie => Ok(false),
+                ProcessStatus::Tracing => Ok(false),
+                ProcessStatus::Dead => Ok(false),
+                ProcessStatus::Wakekill => Ok(false),
+                ProcessStatus::Parked => Ok(false),
+                ProcessStatus::LockBlocked => Ok(false),
+                ProcessStatus::Unknown(_s) => Ok(false),
```

Comment:
maybe this would be a little clearer

```rust
use ProcessStatus::*;
let is_running = matches!(process.status(), Idle | Run | Sleep | Waking);
```